### DISCUSSION TITLE
Incorporation of SEB Application Code Signatures into `X-SafeExamBrowser-RequestHash` HTTP Header

### DISCUSSION THREAD

**Discussion by mohammad-aliB** [SafeExamBrowser/seb-mac#337] [Q&A]
The browser exam key feature documentation [SEB-Specification-BrowserExamKey](https://safeexambrowser.org/developer/documents/SEB-Specification-BrowserExamKey.pdf) states the following "That’s why the base for a Browser Exam Key is a code signature of the whole SEB application (binaries and resources)"

Reviewing the codebase, the generation of the Browser Exam Key occurs in the `SEBBrowserController.m` file, specifically in the [`browserExamKeyForURL` class](https://github.com/SafeExamBrowser/seb-mac/blob/7ce7b2d310074cdffd0c26904f6c78f81ad7e404/Classes/BrowserComponents/SEBBrowserController.m#L537). This class makes use of the `self.browserExamKey` variable, which is retrieved from the `org_safeexambrowser_currentData` preference.

In the [`SEBCryptor.m` file](https://github.com/SafeExamBrowser/seb-mac/blob/main/Classes/Cryptography/SEBCryptor.m), a dictionary is filtered to only include keys prefixed with `org_safeexambrowser_SEB_` ([Lines 254-280](https://github.com/SafeExamBrowser/seb-mac/blob/7ce7b2d310074cdffd0c26904f6c78f81ad7e404/Classes/Cryptography/SEBCryptor.m#L254)). A new Browser Exam Key is then generated using a new salt in combination with this filtered list, as seen in the [`checksumForPrefDictionary` method](https://github.com/SafeExamBrowser/seb-mac/blob/7ce7b2d310074cdffd0c26904f6c78f81ad7e404/Classes/Cryptography/SEBCryptor.m#L306).

At which point in this process are the code signatures of the SEB application calculated and how are they incorporated into the `X-SafeExamBrowser-RequestHash` HTTP header?

**Comment by danschlet** [comment]
Calculating and incorporating the code signatures of the SEB application into the BEK is not part of the publicly available open source code of SEB.
